python/lsst/ts/observatory/control/base_calsys.py

```diff
@@ -188,6 +205,25 @@ def load_calibration_config_file(self, filename: str | None = None) -> None:
         with open(data_path, "r") as f:
             self.calibration_config = yaml.safe_load(f)
 
+        config_schema = None
+        try:
+            with open(schema_path, "r") as f:
+                config_schema = yaml.safe_load(f)
+        except FileNotFoundError:
+            self.log.warning(
+                f"Configuration schema file {schema_path} not found. "
+                f"No schema validation is performed for {data_path}."
+            )
+
+        if config_schema:
+            for item in self.calibration_config:
+                try:
```

Comment:
I don't think re-raising the exception like this is adding anything useful. What you could do instead is try to validate all sessions and then raise an exception at the end aggregating all the failures.

Something like this:

```
validation_errors = ""
for item in self.calibration_config:
    try:
        # validate config
        ...
    except jsonschema.ValidationError as e:
        validation_errors += f"\t{item} failed validation: {e.message}.\n"
        self.log.exception(f"{item} failed validation.")
if validation_errors:
    raise RuntimeError(f"Failed schema validation:\n{validation_errors}Check logs for more information.")
```